server/src/middleware/verify-token.middleware.ts

```diff
@@ -0,0 +1,47 @@
+import { NextFunction, Request, Response } from 'express';
+import jwt from 'jsonwebtoken';
+import { errorHandler } from '../utils';
+import { HttpMethods, Separators, StatusCode, messages } from '../constants';
+import { JWT_SECRET } from '../public.env';
+import { UserEntity } from '../resources/user/user.interfaces';
+
+const authorizationCallBack = async (
+  req: Request,
+  res: Response,
+  next: NextFunction,
+): Promise<Response | undefined | void> => {
+  if (req.method === HttpMethods.OPTIONS) {
+    return next();
+  }
+
+  const auth = req.headers.authorization as string;
+
+  if (!auth) {
+    return res.status(StatusCode.UNAUTHORIZED).json({ message: messages.notAuthorized });
+  }
+
+  // authorization = `Bearer ${token}`
+  const [tokenType, token] = auth.split(Separators.SPACE);
+
+  if (tokenType !== 'Bearer') {
+    return res.status(StatusCode.FORBIDDEN).json({ message: messages.invalidToken });
+  }
+
+  const user = jwt.verify(token, JWT_SECRET) as UserEntity;
+  req.user = user;
```

Comment:
According to AC5:
Authentication middleware is added to check if token provided is valid **and if user encoded in token exists**

Reply:
Done.